sdk/go/common/resource/properties_path.go

```diff
@@ -299,6 +301,148 @@ func (p PropertyPath) Contains(other PropertyPath) bool {
 	return true
 }
 
+func (p PropertyPath) reset(old, new PropertyValue) bool {
```

Comment:
This logic is quite difficult to follow. I'd attribute that to the number of conditions a reader has to keep in their head simultaneously.

Reply:
The logic was difficult to write 😆 I wanted to write a few more tests to cover this because it's gnarly but haven't had time today. I'll give it a fresh look when I get back to this, might be some way to clean it up.